### PR TITLE
feat(encoding): add Best compression level (zstd level 11, btlazy2 strategy)

### LINKED ISSUE #7: feat: implement Best compression level (zstd level 11, btlazy2 strategy)
## Summary

`CompressionLevel::Best` currently triggers `unimplemented!()`. Level 11 uses binary tree with lazy matching — highest compression ratio in the current enum.

### C reference implementation (zstd level 11)
- **Strategy**: `ZSTD_btlazy2` (binary tree + lazy2)
- **Parameters**: `windowLog=24, chainLog=25, hashLog=24, searchLog=7, minMatch=3, targetLength=128`
- **Key file**: `lib/compress/zstd_lazy.c` (ZSTD_insertDUBT1, ZSTD_DUBT_findBestMatch)

### Algorithm
1. **Binary search tree** — sorted by suffix content for O(log n) match finding
2. **DUBT (Doubly-Unsorted Binary Tree)** — insertion marks entries as sorted/unsorted (`ZSTD_DUBT_UNSORTED_MARK = 1`)
3. **Common-length optimization** — `commonLengthSmaller/Larger` to skip already-compared bytes during tree traversal
4. **Lazy2 on top** — same 2-step look-ahead as level 7 but with btree match finder instead of hash chains

### What needs to be implemented

1. **Binary tree data structure** — left/right pointers in chain table, sorted insertion
2. **DUBT insertion** — `insertDUBT1` equivalent with sorted/unsorted mark handling
3. **Tree search** — `ZSTD_DUBT_findBestMatch` with common-length caching
4. **Integration with lazy2** — same lazy evaluation but better matches from btree

### Acceptance criteria
- [ ] `CompressionLevel::Best` no longer panics
- [ ] Roundtrip + interop tests pass
- [ ] Compression ratio within 10% of C zstd level 11 on silesia corpus
- [ ] Noticeably better ratio than Better level

### Dependencies
- #6 (Better level) — shared lazy matching infrastructure

## Time estimate
4d

## Blocked by
- #6 (Better level — shared lazy matching infrastructure)

### PR DESCRIPTION
**Pull request by polaz**
## Summary

- Implement `CompressionLevel::Best` — roughly equivalent to zstd level 11
- Uses enhanced hash-chain matcher with deep lazy2 matching: 16 MiB window, 2M/1M hash/chain tables, 32-candidate search depth, 128 target match length
- HC matcher parameters (`hash_log`, `chain_log`, `search_depth`, `target_len`) are now configurable per level via `configure()` method
- Full streaming encoder and frame compressor support
- CLI level `4` maps to Best

## Compression ratio vs C zstd

| Dataset | Input | Better | Best | C-L7 | C-L11 | Best/C11 | Better/C7 |
|---------|-------|--------|------|------|-------|----------|-----------|
| corpus z000033 | 1.0M | 541,592 | 540,153 | 510,030 | 509,317 | 106.1% | 106.2% |
| huffman 4M | 4.0M | 2,957,814 | 2,959,960 | 2,949,731 | 2,949,773 | 100.3% | 100.3% |
| log lines 4M | 4.0M | 582 | 580 | 526 | 866 | **67.0%** | 110.6% |
| log lines 16M | 16M | 1,830 | 1,828 | 1,684 | 3,077 | **59.4%** | 108.7% |

- Best/C-L11 gap (6.1%) matches Better/C-L7 gap (6.2%) — systematic encoder pipeline gap
- On structured log data, Best **beats** C-L11 by 1.5–1.7× (C btlazy2 degrades on repetitive data)
- Improvement Better→Best (0.27% on corpus) exceeds C-L7→C-L11 improvement (0.14%)

## Test plan

- [x] 11 roundtrip integrity tests (compressible, random, multi-block, streaming, edge cases, repeat offsets, large window)
- [x] Cross-validation: Rust Best → C FFI decompress on corpus
- [x] Ratio assertion: Best < Better (strict) on corpus proxy
- [x] Large-window regression: 9 MiB gap between duplicated regions (Best beats Better)
- [x] Streaming encoder roundtrip at Best level
- [x] CLI level 4 parsing and compression
- [x] All 204 tests pass, zero warnings

Closes #7

## Summary by CodeRabbit

* **New Features**
  * "Best" compression level is now supported in the CLI and library for frame and streaming compression (accessible via the CLI level option), offering higher compression with a larger window and deeper matching (≈ zstd level 11).

* **Documentation**
  * README and benchmark docs updated to mark Best implemented and to describe its behavior and limits.

* **Tests**
  * Added/expanded tests for Best roundtrips, streaming correctness, cross-language decompression, and comparative compression quality.

### CODE REVIEW

File: zstd/src/tests/roundtrip_integrity.rs [hunk 1]
```diff
@@ -467,3 +475,104 @@ fn roundtrip_better_level_large_window() {
...
+fn roundtrip_best_level_repeat_offsets() {
+    let data = repeat_offset_fixture(b"ABCDE12345", 10_000);
+    assert_eq!(roundtrip_best(&data), data);
+}
+
+#[test]
+fn roundtrip_best_level_large_literals() {
+    let data = generate_huffman_friendly(300, 128 * 1024, 64);
+    assert_eq!(roundtrip_best(&data), data);
+}
```
[the remark this answers is not in the document]
Comment (polaz) [reply zstd/src/tests/roundtrip_integrity.rs]: @coderabbitai Won't fix — this mirrors the established pattern from the Better level PR (#48). Each level's tests use different seeds and data sizes intentionally. Extracting a macro/helper for 6 one-liner tests would add indirection without meaningful DRY benefit. If a third or fourth level is added, a test matrix macro would become worthwhile.
